File: tests/fixtures/negative/security/ProcessBuilderShellArg.java
package test;

class ProcessBuilderShellArgJavaSafeFixture {
    void grep(String pattern) throws java.io.IOException {
        new ProcessBuilder("grep", pattern, "/var/log/app.log").start();
        new ProcessBuilder("sh", "-c", "grep fixed /var/log/app.log").start();
    }
}

File: tests/fixtures/negative/security/ProcessBuilderShellArg.kt
package test

class ProcessBuilderShellArgSafeFixture {
    fun grep(pattern: String) {
        ProcessBuilder("grep", pattern, "/var/log/app.log").start()
        ProcessBuilder("sh", "-c", "grep fixed /var/log/app.log").start()
        ProcessBuilder("sh", "-lc", "grep $pattern /var/log/app.log").start()
    }
}
